### PR TITLE
Clean up CaptureSync

### PR DESCRIPTION
**Pull request by pierricgimmig**
CaptureClient::Capture was taking over 20 arguments and was 
duplicated by CaptureClient::CaptureSync which made adding 
a new capture option quite annoying. Clean up that code a little.

Tests: profiled on instance with manual and dynamic instrumentation.
b/187170164

### CODE REVIEW

**Review by beckerhe — APPROVED** [review]
That's a nice change. Looks good to me.

I'm only wondering if `ClientCaptureOptions` deserves its own header and cpp file. WDYT?

File: src/CaptureClient/include/CaptureClient/CaptureClient.h [hunk 1]
```diff
@@ -32,6 +32,34 @@
 
 namespace orbit_capture_client {
 
+struct CaptureClientOptions {
```
Comment (dpallotti) [inline src/CaptureClient/include/CaptureClient/CaptureClient.h:35]: The word order is strange, I think this should be called `ClientCaptureOptions` (the name @beckerhe used without noticing it's different, and also as counterpart to `ToGrpcCaptureOptions`).

File: src/FakeClient/FakeClientMain.cpp [hunk 2]
```diff
@@ -293,63 +294,64 @@ int main(int argc, char* argv[]) {
...
-  ORBIT_FAIL_IF(process_id == 0, "PID to capture not specified");
+  ORBIT_LOG("process_id=%d", options.process_id);
+  ORBIT_FAIL_IF(options.process_id == 0, "PID to capture not specified");
 
-  uint16_t samples_per_second = absl::GetFlag(FLAGS_sampling_rate);
-  ORBIT_LOG("samples_per_second=%u", samples_per_second);
-  constexpr uint16_t kStackDumpSize = 65000;
-  const UnwindingMethod unwinding_method =
+  options.samples_per_second = absl::GetFlag(FLAGS_sampling_rate);
+  ORBIT_LOG("samples_per_second=%f", options.samples_per_second);
```
Comment (dpallotti) [inline src/FakeClient/FakeClientMain.cpp:309]:
```suggestion
  ORBIT_LOG("samples_per_second=%.0f", options.samples_per_second);
```
The flag is integer anyway.

File: src/CaptureClient/CaptureClient.cpp [hunk 3]
```diff
@@ -45,18 +45,73 @@ using UnwindingMethod = orbit_grpc_protos::CaptureOptions::UnwindingMethod;
...
+
+  auto api_functions = FindApiFunctions(module_manager);
+  *(capture_options.mutable_api_functions()) = {api_functions.begin(), api_functions.end()};
+
+  return capture_options;
+}
+
+}  // namespace
+
 // TODO(b/187170164): This method contains a lot of arguments. Consider making it more structured.
```
Comment (dpallotti) [inline src/CaptureClient/CaptureClient.cpp:109]: I guess this can be removed then.

**Review by dpallotti — APPROVED** [review]
Looks good otherwise.

File: src/FakeClient/FakeClientMain.cpp (hunk 2 above)
Comment (pierricgimmig) [reply src/FakeClient/FakeClientMain.cpp]: Done, thanks.

File: src/CaptureClient/CaptureClient.cpp (hunk 3 above)
Comment (pierricgimmig) [reply src/CaptureClient/CaptureClient.cpp]: Indeed :-), removed, thanks.

**Comment by pierricgimmig** [comment]
> I'm only wondering if `ClientCaptureOptions` deserves its own header and cpp file. WDYT?

Good idea, done, thanks.

**Pull request merged**